client/assets/calendar.js

```diff
@@ -0,0 +1,121 @@
+	const ics_day_list = ['SU', 'MO', 'TU', 'WE', 'TH', 'FR', 'SA'] //days of the week for ICS standard
+	const semester_year = (plan.courses[0].dates[0].slice(0, 2) < moment().month()) ? moment().year() + 1 : moment().year()
+
+	//required iCalendar info
+	//calendar_string is an array because what we use to print requires an array
+	let calendar_string = ['BEGIN:VCALENDAR\nPRODID:-//Nick and Pedram, Squaaaad//Course Calendar//EN\nVERSION:2.0\nCALSCALE:GREGORIAN']
+	calendar_string += '\nMETHOD:PUBLISH\nX-WR-CALNAME:' + plan.title
+	//timezone and daylight savings info
+	calendar_string += '\nX-WR-TIMEZONE:America/Los_Angeles\nBEGIN:VTIMEZONE\nTZID:America/Los_Angeles'
+	calendar_string += '\nX-LIC-LOCATION:America/Los_Angeles\nBEGIN:DAYLIGHT\nTZOFFSETFROM:-0800\nTZOFFSETTO:-0700\nTZNAME:PDT'
+	calendar_string += '\nDTSTART:19700308T020000\nRRULE:FREQ=YEARLY;BYMONTH=3;BYDAY=2SU\nEND:DAYLIGHT\nBEGIN:STANDARD\nTZOFFSETFROM:-0700'
+	calendar_string += '\nTZOFFSETTO:-0800\nTZNAME:PST\nDTSTART:19701101T020000\nRRULE:FREQ=YEARLY;BYMONTH=11;BYDAY=1SU\nEND:STANDARD'
+	calendar_string += '\nEND:VTIMEZONE\n'
+
+	plan.courses.forEach(function(plan_course) {
+		//need to use moments to deal with the bug with the start day of the semester not being first day of the course
+		//start_moment = the starting date of the course and time the course starts each day
+		let start_moment = moment(plan_course.dates[0] + '/' + semester_year)
+		//end_moment = the ending date of the course along with the time this course ends each day
+		let end_moment = moment(plan_course.dates[1] + '/' + semester_year)
+
+		start_moment.hour(Number(plan_course.times[0].slice(0,2)))
+		start_moment.minute(Number(plan_course.times[0].slice(3,5)))
+		end_moment.hour(Number(plan_course.times[1].slice(0,2)))
+		end_moment.minute(Number(plan_course.times[1].slice(3,5)))
+
+		//adjust for 24 hr time
+		if (plan_course.times[0].slice(-2) == 'pm' && start_moment.hours() != 12) {
+			start_moment.add(12, 'hours')
+		}
+		if (plan_course.times[1].slice(-2) == 'pm' && end_moment.hours() != 12) {
+			end_moment.add(12, 'hours')
+		}
+
+		//create an array that maps the days proveded by banner to the days needed by ICS standard
+		let course_dows = []
+		plan_course.days.forEach(function(day) {
+			switch(day) {
+				case 'M':
+				course_dows.push(1)
+			break;
+				case 'T':
+				course_dows.push(2)
+			break;
+				case 'W':
+				course_dows.push(3)
+			break;
+				case 'R':
+				course_dows.push(4)
+			break;
+				case 'F':
+				course_dows.push(5)
+			break;
+				case 'S':
+				course_dows.push(6)
+			break;
+				case 'U':
+				course_dows.push(0)
+			break;
+		}
+		})
+
+		//there is an issue with the first day of the semester being later than when the first day of this class is
+		while(!course_dows.includes(start_moment.day())) {
+		start_moment.add(1, 'day')
+		}
+
+		let right_now = (moment().year()).toString() //used to show when this was made in correct ICS format
```

Comment:
If you want moment to output into a format use the [.format()](https://momentjs.com/docs/#/displaying/format/) function I believe this may even be the format you are looking for here: [ISO](https://momentjs.com/docs/#/displaying/as-iso-string/)

This code is way too complicated for what it is actually doing